dali/python/nvidia/dali/external_source.py

```diff
@@ -155,81 +155,99 @@ class ExternalSource():
+    yielded by the function is requested.
 
 `name` : str, optional
-    The name of the data node - used when feeding the data in ``iter_setup``; can be omitted if
+    The name of the data node.
+
+    Used when feeding the data in ``iter_setup`` and can be omitted if
     the data is provided by ``source``.
 
+
 `layout` : :ref:`layout str<layout_str_doc>` or list/tuple thereof
-    If provided, sets the layout of the data. When ``num_outputs > 1``, layout can be a list
-    containing a distinct layout for each output. If the list has fewer elements than ``num_outputs``,
-    only the first outputs have the layout set, the reset have it cleared.
-
-`cuda_stream` : optional, `cudaStream_t` or an object convertible to `cudaStream_t`, e.g. `cupy.cuda.Stream`, `torch.cuda.Stream`
-    The CUDA stream, which is going to be used for copying data to GPU or from a GPU
-    source. If not set, best effort will be taken to maintain correctness - i.e. if the data
-    is provided as a tensor/array from a recognized library (CuPy, PyTorch), the library's
-    current stream is used. This should work in typical scenarios, but advanced use cases
-    (and code using unsupported libraries) may still need to supply the stream handle
-    explicitly.
-
-    Special values:
-      *  0 - use default CUDA stream
-      * -1 - use DALI's internal stream
-
-    If internal stream is used, the call to ``feed_input`` will block until the copy to internal
-    buffer is complete, since there's no way to synchronize with this stream to prevent
-    overwriting the array with new data in another stream.
+    If provided, sets the layout of the data.
+
+    When ``num_outputs > 1``, the layout can be a list that contains a distinct layout for each
+    output. If the list has fewer than ``num_outputs`` elements, only the first
+    outputs have the layout set, the rest of the outputs don't have a layout set.
+
+`cuda_stream` : optional, ``cudaStream_t`` or an object convertible to ``cudaStream_t``, such as
+    ``cupy.cuda.Stream`` or ``torch.cuda.Stream``
+    The CUDA stream is used to copy data to a GPU or from the GPU source.
+
+    If this parameter is not set, a best-effort will be taken to maintain correctness. That is,
+    if the data is provided as a tensor/array from a recognized library such as CuPy or PyTorch,
+    the library's current stream is used. Although this process works in typical scenarios,
+    with advanced use cases, and code that uses unsupported libraries, you might need to
+    explicitly supply the stream handle.
+
+    Here are the special values:
+      *  0 - Use the default CUDA stream
+      * -1 - Use DALI's internal stream
+
+    Since you cannot synchronize with this stream to prevent overwriting the array with new data
+    in another stream, if an internal stream is used, the call to ``feed_input`` will block
+    until the copy to the internal buffer is complete.
```

Comment:
This is very hard to understand compared to what was there before. Revert?

Reply:
Done